earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -18,19 +18,131 @@
 #include <unistd.h>
 #include <sys/syscall.h>
 #include <time.h>
-
+#include <cstdlib>
 #include "common/performancelogger.h"
 #include "common/notify.h"
 #include "common/timeutils.h"
+#include <cassert>
 
 using namespace std;
 using namespace getime;
 
-// Initialize static members of Profiler class
-PerformanceLogger * const PerformanceLogger::_instance = new PerformanceLogger();
+string ioFileName;
+string threadFileName;
+string timeFileName;
+
+void generateFileNames() {
+    // needs mutex lock
+    time_t t = time(0);
+    tm date = *localtime(&t);
+    char buf[256];
+    std::fill(buf,buf+256,0);
```

Comment:
Q: Is this necessary?  Doesn't `strftime` already add a terminating `'\0'` character?

Reply:
You can use `sizeof(buf)` instead of `256`.

Reply:
I like to err on the side of caution. sizeof would be ok, but we already know the size here so I figured just use it. I suppose it comes down to standard convention.

Reply:
This is minor, but, the documentation of `strftime` says:

> `size_t strftime( char *restrict str, size_t count, const char *restrict format, const struct tm *restrict time );`
> Converts the date and time information from a given calendar time time to a **null-terminated** multibyte character **string str** according to format string format. Up to count bytes are written.

[empasis added]

It's intended to be used with an uninitialized buffer.  I think there may be compiler flags to initialize buffers (which often already happens).

Reply:
Fair enough. Changing.